src/tiled/tilesetdock.cpp

```diff
@@ -939,3 +942,17 @@ void TilesetDock::swapTiles(Tile *tileA, Tile *tileB)
     QUndoStack *undoStack = mMapDocument->undoStack();
     undoStack->push(new SwapTiles(mMapDocument, tileA, tileB));
 }
+
+void TilesetDock::replaceSelectedObjectsTile(Tile *tile)
+{
+    if (!mMapDocument)
+        return;
+
+    auto &selectedObjects = mMapDocument->selectedObjects();
+
+    if (selectedObjects.isEmpty())
+        return;
+
+    QUndoStack *undoStack = mMapDocument->undoStack();
+    undoStack->push(new ReplaceObjectsTile(mMapDocument, selectedObjects, tile));
```

Comment:
Not all selected objects may be tile objects. I would suggest to only support this action on existing tile objects, because the behavior when setting a tile on a rectangle, circle, polygon or text object isn't well defined (or at least, this is currently not a supported scenario).